src/app/private/containers/goal/pages/HealthGoalDetail.tsx

```diff
@@ -41,6 +41,13 @@ const HealthGoalDetail: React.FC = () => {
     navigate('/health-goal');
   };
 
+  const handleClickViewDataAnalyzation = async () => {
+    await activateHealthGoal({
+      healthGoalId: id as string,
+    });
+    navigate('/health-goal');
```

Comment:
@Ellen-ho looks like you are coping the code from other function. Suggest to remove that part and add a TODO comments